### PR TITLE
fix when minimizer is fn

### PR DESCRIPTION
**Pull request by hulkish**
As discussed offline with @sokra, this is a fix for when you provide a function entry for `optimization.minimizer` config.

**What kind of change does this PR introduce?**

bugfix (related: https://github.com/webpack/webpack/pull/8405)

**Did you add tests for your changes?**

No, should i?

**Does this PR introduce a breaking change?**

No

**What needs to be documented once your changes are merged?**

Done: https://github.com/webpack/webpack.js.org/pull/2743

### CODE REVIEW

File: lib/WebpackOptionsApply.js [hunk 1]
```diff
@@ -450,7 +450,7 @@ class WebpackOptionsApply extends OptionsApply {
 		if (options.optimization.minimize) {
 			for (const minimizer of options.optimization.minimizer) {
 				if (typeof minimizer === "function") {
-					minimizer.apply(compiler);
+					minimizer.call(compiler, compiler);
```
Comment (hulkish) [inline lib/WebpackOptionsApply.js]: maybe should be bound to null, since we're already providing compiler?

**Comment by sokra** [comment]
Needs a test case

**Comment by hulkish** [comment]
@sokra should i rebase this pr to point to master? I noticed same issue in master branch.

**Comment by sokra** [comment]
> @sokra should i rebase this pr to point to master? I noticed same issue in master branch.

yep

**Comment by hulkish** [comment]
ok, should be all set now.

**Comment by sokra** [comment]
Thanks

**Comment by hulkish** [comment]
@sokra will this be cherry picked to next branch?

**Comment by sokra** [comment]
I merge master into next on every release

**Pull request merged**